### PR TITLE
Issue #68 Use virt-sparsify on created qcow2 image.

### PR DESCRIPTION
**Pull request by praveenkumar**
virt-sparsify can be used to create vm disk sparse. It is
reducing around ~700MB disk size and also help us to have
smaller tarball.

### CODE REVIEW

File: createdisk.sh [hunk 1]
```diff
@@ -54,6 +54,10 @@ function create_qemu_image {
     ${QEMU_IMG} create -o lazy_refcounts=on -f qcow2 $destDir/${CRC_VM_NAME}.qcow2 31G
     ${VIRT_RESIZE} --expand /dev/sda3 $destDir/${VM_PREFIX}-base $destDir/${CRC_VM_NAME}.qcow2
 
+    # Make sure you have enough space to tmp to run VIRT_SPARSIFY
+    # Read limitation section of `man virt-sparsify`.
+    ${VIRT_SPARSIFY} --in-place $destDir/${CRC_VM_NAME}.qcow2
```
Comment (cfergeau) [inline createdisk.sh]: Previous code used to set TMPDIR to the current dir:
```
    # TMPDIR must point at a directory with as much free space as the size of the
    # image we want to sparsify
    TMPDIR=$(pwd)/$destDir ${VIRT_SPARSIFY} $destDir/${VM_PREFIX}-base $destDir/${CRC_VM_NAME}.qcow2
    rm -fr $destDir/.guestfs-*
```
https://github.com/code-ready/snc/commit/f0768092dc2f0ffc78853dc2165ff109f0e5a0bf

File: createdisk.sh [hunk 2]
```diff
@@ -228,7 +228,7 @@ ${SSH} core@api.${CRC_VM_NAME}.${BASE_DOMAIN} -- 'sudo crictl stopp $(sudo crict
 ${SSH} core@api.${CRC_VM_NAME}.${BASE_DOMAIN} -- 'sudo crictl rmp $(sudo crictl pods -q)'
```
Comment (cfergeau) [inline createdisk.sh]: Maybe commit log should be "We are not using sudo to remove the `config.json` file from `/etc/kubernetes`. This silently fails, and the resulting bundle contains this file."

File: createdisk.sh (hunk 1 above)
Comment (praveenkumar) [reply createdisk.sh]: @cfergeau right but right now we are using the `--inplace` which doesn't work with `tmp` dir. I did try using `--tmp` option along with `--in-place` and that error out.

Comment (cfergeau) [reply createdisk.sh]: setting `TMPDIR` should not error out. However, 
`If you are using the --in-place option, then large amounts of temporary space are not required.` so I'm not sure the comment you added is needed at all?

Comment (praveenkumar) [reply createdisk.sh]: @cfergeau I just put `enough space` not large amount which you are right required only if we don't use the `inplace` option.

Comment (cfergeau) [reply createdisk.sh]: "to tmp" -> "in /tmp" if you want to keep the comment

File: createdisk.sh [hunk 3]
```diff
@@ -10,7 +10,7 @@ SCP="scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i id_rsa_c
 OC=${OC:-oc}
 
 function get_git_tag {
-    GIT_TAG=$(git describe --exact-match HEAD) || GIT_TAG=
+    GIT_TAG=$(git describe --exact-match --tags HEAD) || GIT_TAG=
```
Comment (cfergeau) [inline createdisk.sh]: Why don't  we sign our tags though? :)

File: createdisk.sh (hunk 2 above)
Comment (cfergeau) [reply createdisk.sh]: commit log does not seem to have been adjusted in the latest version?

File: createdisk.sh (hunk 1 above)
Comment (gbraad) [reply createdisk.sh]: I can not judge about this, as this is not something I can test now

File: createdisk.sh (hunk 2 above)
Comment (praveenkumar) [reply createdisk.sh]: @cfergeau it's updated :) https://github.com/code-ready/snc/pull/71/commits/4d78be025616ae861e60bf22c2439ac9f6bde4e4

**Pull request merged**